src/contract.rs

```diff
@@ -377,6 +391,63 @@ fn is_executable(ask_order: &AskOrderV2, bid_order: &BidOrderV2) -> bool {
     ask_base == bid_base && ask_quote == bid_quote
 }
 
+/// Switches the scope's current owner value to the given owner value.
+/// If replace_value_owner is specified, the new_owner value will also be used as the value owner.
+/// If validate_expected_owner is given an address, the owner's current address will be verified against it.
+fn replace_scope_owner(
+    mut scope: Scope,
+    new_owner: Addr,
+    replace_value_owner: bool,
+    validate_expected_owner: Option<Addr>,
+) -> Result<Scope, ContractError> {
+    let owners = scope
+        .owners
+        .iter()
+        .filter(|owner| owner.role == PartyType::Owner)
+        .collect::<Vec<&Party>>();
+    // if more than one owner is specified, removing all of them can potentially cause data loss
+    if owners.len() != 1 {
+        return Err(ContractError::InvalidScopeOwner {
+            scope_address: scope.scope_id,
+            explanation: format!(
+                "the scope should only include a single owner, but found: {}",
+                owners.len(),
+            ),
+        });
+    }
+    // The owner of the scope must be the sender to the contract - otherwise, scopes could be released on the behalf of others
+    if let Some(expected_owner) = validate_expected_owner {
+        let owner = owners.first().unwrap();
+        if owner.address != expected_owner {
+            return Err(ContractError::InvalidScopeOwner {
+                scope_address: scope.scope_id,
+                explanation: format!(
+                    "the scope owner was expected to be [{}], not [{}]",
+                    expected_owner, owner.address,
+                ),
+            });
+        }
+    }
+    // Empty out all owners from the scope now that it's verified safe to do
+    scope.owners = scope
+        .owners
+        .into_iter()
+        .filter(|owner| owner.role == PartyType::Owner)
```

Comment:
shouldn't this be `!=` ?

Reply:
LOL yup you are correct sir.  Yet another reason why that metadata params is gonna be a very good thing to have exposed - I did not test this piece because I couldn't

Reply:
:truestory: